runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -223,6 +224,10 @@ class ROMClassCreationContext
 	bool isClassAnon() const { return J9_FINDCLASS_FLAG_ANON == (_findClassFlags & J9_FINDCLASS_FLAG_ANON); }
 	bool alwaysSplitBytecodes() const { return J9_ARE_ANY_BITS_SET(_bctFlags, BCT_AlwaysSplitBytecodes); }
 
+	bool shouldInjectIdentityObjectInterface(J9CfrClassFile *classfile) const {
+		return (J9_ARE_NO_BITS_SET(classfile->accessFlags, CFR_ACC_VALUE_TYPE|CFR_ACC_INTERFACE|CFR_ACC_ABSTRACT) && (classfile->majorVersion < VALUE_TYPES_MAJOR_VERSION));
```

Comment:
Right now Im only checking to see that it is a concrete type, however, in a real solution I think I would need to check if the superclass already implements IdentityObject before adding it to the current class.

Reply:
I don't know that we're guaranteed that every class with `classfile->majorVersion >= VALUE_TYPES_MAJOR_VERSION` will correctly implement IdentityObject/InlineObject as user-created bytecode generators may not have been updated even though they have a new classfile version

Reply:
The parent ROMClass may not have been created yet - we may get a class load event for class `B`, and create the ROMClass at that point.  As part of creating the RAM class, we'll trigger the load of the super class `A`, etc.

Reply:
`CFR_ACC_VALUE_TYPE` will this exist?  It wasn't clear to me if there would be a spec classfile bit or we'd have to look for the whether this class implements the `InlineObject` interface